src/api/meal_plan.py

```diff
@@ -1,89 +1,227 @@
+            
+            validate_meal_recipe = connection.execute(text("""SELECT 1 FROM plans_recipes WHERE meal_plan_id = :meal_plan_id AND recipe_id = :recipe_id"""), 
+                                                      {"meal_plan_id": meal_plan_id, "recipe_id": recipe_id}).one_or_none()
+            
+            if validate_meal_recipe is None:
+                raise Exception("Meal plan does not contain recipe.")
+            
+            connection.execute(text("""DELETE FROM plans_recipes WHERE meal_plan_id = :meal_plan_id AND recipe_id = :recipe_id"""), {"meal_plan_id": meal_plan_id, "recipe_id": recipe_id})
+
+            return Response(status_code=200)
+        
+    except Exception as e:
+        raise HTTPException(status_code=404, detail=f"Error adding recipe: {e}")
+
+@router.post("/{meal_plan_id}/share")
+def share_meal_plan(meal_plan_id: int, author_id: int, recipient_id: int):
     with db.engine.begin() as connection:
-        connection.execution_options(isolation_level="REPEATABLE READ")
-        result = connection.execute(text("""
-            SELECT COALESCE(
-                SUM(
-                    ri.quantity * COALESCE(
-                        CAST(m.energy_amount AS double precision),
-                        u.energy_amount,
-                        un.energy_amount
-                    )
-                ),
-                0
-            ) AS total_calories
-            FROM meal_logs ml
-            JOIN recipe_ingredients ri ON ml.recipe_id = ri.recipe_id
-            LEFT JOIN menustat m ON ri.ingredient_id = m.menustat_id
-            LEFT JOIN usda_branded u ON ri.ingredient_id = u.fdc_id
-            LEFT JOIN usda_non_branded un ON ri.ingredient_id = un.id
-            WHERE ml.meal_plan_id = :meal_plan_id AND ml.date = :date
-        """), {"meal_plan_id": meal_plan_id, "date": date})
+        if author_id is recipient_id:
+            raise HTTPException(status_code=400, detail="Author id and recipient id can not be the same.")
+
+        try:
+            validate_meal_plan = connection.execute(text("""SELECT 1 FROM meal_plans WHERE id = :id AND author_id = :author_id"""), {"id": meal_plan_id, "author_id": author_id}).one_or_none()
+
+            if validate_meal_plan is None:
+                    raise Exception("Meal plan and author id combination not found.")
+            
+            connection.execute(text("""INSERT INTO shared_meal_plans (meal_plan_id, recipient_id) VALUES (:meal_plan_id, :recipient_id)"""), {"meal_plan_id": meal_plan_id, "recipient_id": recipient_id})
+            
+            return Response(status_code=200)
         
-        total_calories = result.fetchone()[0]
-    
-    return {"date": date, "total_calories": total_calories}
+        except Exception as e:
+            raise HTTPException(status_code=404, detail=f"Error sharing plan: {e}")
+        
+@router.get("/{account_id}")
+def get_meal_plans(account_id: int):
```

Comment:
What are your thoughts on moving this to be under `/accounts/{id}/meal-plans/`? Currently, recipes are retrieved for an account via the `/accounts/` endpoint, so I think it would make sense to implement this similarly for consistency.
I think it would also be a little clearer, as if I did `/meal-plans/123`, I would assume it's querying a meal plan with `id` of `123`, rather than the user `123`.

Reply:
I made the move to accounts. Makes sense to me.